src/node/rpc/memberfrontend.h

```diff
@@ -66,6 +66,19 @@ namespace ccf
            nodes->put(id, *info);
            return true;
          }},
+        // retire a node
+        {"retire_node",
+         [this](Store::Tx& tx, const nlohmann::json& args) {
+           const auto id = args;
+           auto nodes = tx.get_view(this->network.nodes);
+           auto info = nodes->get(id);
+           if (!info)
+             throw std::logic_error("Node does not exist.");
```

Comment:
`return jsonrpc::error()` ?

Reply:
we throw a logic_error everywhere else so it makes sense to throw here too though. it get's caught and returned as a jsonrpc error